### ISSUE TITLE
cactus are very fragmented

### ISSUE THREAD

**Issue by maomao2023** [ComparativeGenomicsToolkit/cactus#980]
Hi, glennhickey,

I'm sorry to bother you, but I have a problem that has been troubling me for a long time and I need your help.

I have a whole-genome alignment result for 300 species using cactus, but I found that the alignment results obtained from cactus are very fragmented. This has caused some difficulties in processing because I have some lineage-specific conserved sequences obtained from lastz based on the reference species, but these lineage-specific conserved sequences are divided into multiple fragments in the cactus result, making it impossible for me to correspond the previous lastz results with the current cactus results. 

For example, I obtained a 100bp lineage-conserved sequence from the axt file obtained from lastz after chaining, but this sequence is divided into 5 fragments in the cactus result, with fragment lengths ranging from 1bp to 19bp. 

I would like to ask if there are any subsequent processing steps for the results of cactus to link adjacent fragments in order to obtain longer alignment results and achieve my goal of using cactus whole-genome alignments to find more lineage-specific conserved sequences.

**Comment by tli71193** [comment]
@glennhickey i have also experienced a lot of fragmentation when performing cactus halLiftover alone. I am trying to implement the chaining strategy you recommended but am trying to figure out best strategies. 
in my experience running the halLiftover, pslPosTarget, and axtChain commands to create chain files for say mouse to humans and using UCSC liftover with the generated chain and a test bed file is highly off compared to using the chain files provided by UCSC. can you explain why this is the case? 

I am wondering if replacing halSynteny would be a better option. can you comment on the difference between halSynteny vs halLiftover when generating the chain files as well?

**Comment by glennhickey** [comment]
I think `halSynteny` starts with `halLiftover` then does some extra filtering to produce cleaner chains. Hopefully @diekhans can shed some more light.

**Comment by tli71193** [comment]
i'm assuming on UCSC has the cactus assembly hub included for hg38 so given this link ([hg38_example](https://genome.ucsc.edu/s/tli824/hg38_example) where I see this region in the hg38 genome and below is the alignment of the mouse genome, the alignment blocks seem very connected. 

But if I do halLiftover for this region, I get around 38 fragments for just this region alone. I am thinking halSynteny would help with this case given that it will produce long alignments rather than the short fragmented blocks compared to just halLiftover.

**Comment by tli71193** [comment]
`liftOver -minMatch=0.5 mm10.bed mm10ToHg38.over.chain.gz hg38_liftOver.bed unlifted.bed
`
in this case I am going mouse to human, so the mm10 equivalent of the session would be chr14:63230939-63231783

**Comment by diekhans** [comment]
Okay, sorry I didn't ask that well.  However are you running
halLiftover?  Are you using it to lift directly or are you using
it to generate chains to lift?

halLiftover directly does fragment.  This approach creates liftOver chains that
will not be as fragmented:

   https://github.com/ComparativeGenomicsToolkit/hal/blob/chaining-doc/doc/chaining-mapping.md

**Comment by tli71193** [comment]
yes i tried the chaining from the link you provided before and it was really off compared to the UCSC liftover using their provided chains.

I was able to do a work around by stitching the halLiftover fragments manually and it had comparable results to UCSC liftover w/ their provided chains.

Thanks for all the help

**Comment by diekhans** [comment]
Hi Tony,

Glad you got it to work.  I was about to suggest a tool UCSC has
to chain the fragments.  It isn't part of the standard browser
tools, and isn't well tested.

liftover is something we really want to improve, so good to have your
observations.

**Comment by tli71193** [comment]
@diekhans if you have the UCSC tool on you I would most gladly be willing to test it as well.

**Comment by diekhans** [comment]
What kind of system are you using?  I can compile it for you, as
the whole Browser tree is annoying to compile.

Also, do you happen to still have the commands you used do the
genomic chaining?  That has generally worked really well.
Perhaps something is wrong with my instructions.

**Comment by tli71193** [comment]
i think im just using the same command information as the chain mapping link.

system i am using is a linux x86_64

**Comment by diekhans** [comment]
Git the simpleChain program to chain together PSL output from
halLiftover:

https://hgwdev.gi.ucsc.edu/~markd/tmp/simpleChain
